plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveSplitManager.java

```diff
@@ -430,25 +446,22 @@ private TableToPartitionMapping getTableToPartitionMapping(ConnectorSession sess
         return mapColumnsByIndex(columnCoercions.buildOrThrow());
     }
 
-    private static boolean isPartitionUsesColumnNames(ConnectorSession session, HiveStorageFormat storageFormat)
+    private static boolean isPartitionUsesColumnNames(ConnectorSession session, Optional<HiveStorageFormat> storageFormat)
     {
-        switch (storageFormat) {
-            case AVRO:
-                return true;
-            case JSON:
-                return true;
-            case ORC:
-                return isUseOrcColumnNames(session);
-            case PARQUET:
-                return isUseParquetColumnNames(session);
-            default:
-                return false;
+        if (storageFormat.isEmpty()) {
+            return false;
         }
+        return switch (storageFormat.get()) {
+            case AVRO, JSON -> true;
+            case ORC -> isUseOrcColumnNames(session);
+            case PARQUET -> isUseParquetColumnNames(session);
+            default -> false;
+        };
     }
 
-    private TableToPartitionMapping getTableToPartitionMappingByColumnNames(SchemaTableName tableName, String partName, List<Column> tableColumns, List<Column> partitionColumns, HiveTimestampPrecision hiveTimestampPrecision)
+    private static TableToPartitionMapping getTableToPartitionMappingByColumnNames(TypeManager typeManager, SchemaTableName tableName, String partName, List<Column> tableColumns, List<Column> partitionColumns, HiveTimestampPrecision hiveTimestampPrecision)
     {
-        ImmutableMap.Builder<String, Integer> partitionColumnIndexesBuilder = ImmutableMap.builder();
+        ImmutableMap.Builder<String, Integer> partitionColumnIndexesBuilder = ImmutableMap.builderWithExpectedSize(partitionColumns.size());
```

Comment:
Should this be in the first commit ?

Reply:
Sure, moved to first commit in the latest revision.